viamonvif/discovery.go

```diff
@@ -328,7 +358,109 @@ func GetMediaInfoFromProfiles(
 			},
 			Codec: string(profile.VideoEncoderConfiguration.Encoding),
 		})
+
+		// Check if the profile has PTZ capabilities
+		if profile.PTZConfiguration.Token == "" {
+			logger.Debugf("Profile %s does not have PTZ capabilities, %s", profile.Name, streamURI.String())
+			continue
+		}
+
+		logger.Debugf("Profile %s has PTZ capabilities, %s", profile.Name, streamURI.String())
+
+		nodes, err := dev.GetPTZNodes(ctx)
+		if err != nil {
+			logger.Warnf("Unable to determine onvif ptz nodes for profile %s: %s, err: %s",
+				profile.Name, streamURI.String(), err)
+			continue
+		}
+
+		ptzCfg := profile.PTZConfiguration
+		nodeToken := string(ptzCfg.NodeToken)
+
+		// find the matching PTZNode
```

Comment:
Add better comment explaining node filtering here.

Also, this logic well return every (profile_token, ptz_node_token) pair. Do we want to only return one result per ptz node token?